### PR TITLE
New resource: tfe_stack

### PR DESCRIPTION
**Pull request by brandonc**
## Description

Adds BETA support for Stacks as tfe_stack

See documentation for details.

NOTE: This should not be merged until the linked go-tfe PR is released.

_Remember to:_

- [ ] _Update the [Change Log](https://github.com/hashicorp/terraform-provider-tfe/blob/main/docs/changelog-process.md)_
- [x] _Update the [Documentation](https://github.com/hashicorp/terraform-provider-tfe/blob/main/docs/changelog-process.md#updating-the-documentation)_

## Testing plan

<details><summary>Details</summary>
<p>

<code>
provider "tfe" {
}

resource "tfe_project" "example" {
  name = "example-project"
  organization = "example-org"
}

resource "tfe_oauth_client" "gh" {
  organization = "example-org"
  name = "gh-example"
  api_url          = "https://api.github.com"
  http_url         = "https://github.com"
  oauth_token      = "ghp_SECRET_TOKEN"
  service_provider = "github"
  organization_scoped = true
}

resource "tfe_stack" "example" {
  name = "example-renamed"
  description = "the worst stack ever"
  project_id = tfe_project.example.id
  
  vcs_repo {
    identifier = "hashicorp-guides/pet-nulls-stack"
    oauth_token_id = tfe_oauth_client.gh.oauth_token_id
    branch = "main"
  }
}

</code>

</p>
</details> 

## External links

_Include any links here that might be helpful for people reviewing your PR. If there are none, feel free to delete this section._

- [go-tfe PR](https://github.com/hashicorp/go-tfe/pull/920)

## Output from acceptance tests

_Please run applicable acceptance tests locally and include the output here. See [testing.md](https://github.com/hashicorp/terraform-provider-tfe/blob/main/docs/testing.md) to learn how to run acceptance tests._

_If you are an external contributor, your contribution(s) will first be reviewed before running them against the project's CI pipeline._

```
$ ENABLE_BETA=1 make testacc TESTARGS='-run TestAccTFEStackResource_basic -v'
TF_ACC=1 TF_LOG_SDK_PROTO=OFF go test $(go list ./... |grep -v 'vendor') -v -run TestAccTFEStackResource_basic -v -timeout 15m
?   	github.com/hashicorp/terraform-provider-tfe	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-tfe/internal/client	(cached) [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-tfe/internal/logging	(cached) [no tests to run]
?   	github.com/hashicorp/terraform-provider-tfe/internal/provider/validators	[no test files]
?   	github.com/hashicorp/terraform-provider-tfe/version	[no test files]
=== RUN   TestAccTFEStackResource_basic
--- PASS: TestAccTFEStackResource_basic (15.41s)
PASS
ok  	github.com/hashicorp/terraform-provider-tfe/internal/provider	16.048s
...
```

### CODE REVIEW

File: internal/provider/stack.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		UpdatedAt: types.StringValue(v.UpdatedAt.Format(time.RFC3339)),
+	}
+
+	if v.Description != "" {
+		result.Description = types.StringValue(v.Description)
+	}
+
+	if v.VCSRepo.GHAInstallationID != "" {
+		result.VCSRepo.GHAInstallationID = types.StringValue(v.VCSRepo.GHAInstallationID)
+	}
```
Comment (sebasslash) [inline internal/provider/stack.go:62]: Let's add a `nil` check for `v.VCSRepo` before attempting to access its fields

File: website/docs/r/stack.html.markdown.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Name of the stack.
+* `project_id` - (Required) ID of the project where the stack should be created.
+* `description` - (Optional) Description of the stack
+* `vcs_repo` - (Required) Settings for the stack's VCS repository.
```
Comment (sebasslash) [inline website/docs/r/stack.html.markdown.md:51]: Minor nit, show all required arguments first.

File: website/docs/r/stack.html.markdown.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+* `identifier` - (Required) A reference to your VCS repository in the format `<vcs organization>/<repository>` where `<vcs organization>` and `<repository>` refer to the organization and repository in your VCS provider. The format for Azure DevOps is `<ado organization>/<ado project>/_git/<ado repository>`.
+* `branch` - (Optional) The repository branch that Terraform will execute from. This defaults to the repository's default branch (e.g. main).
+* `github_app_installation_id` - (Optional) The installation id of the Github App. This conflicts with `oauth_token_id` and can only be used if `oauth_token_id` is not used.
+* `oauth_token_id` - (Optional) The VCS Connection (OAuth Connection + Token) to use. This ID can be obtained from a `tfe_oauth_client` resource. This conflicts with `github_app_installation_id` and can only be used if `github_app_installation_id` is not used.
+
+## Attributes Reference
+
+* `id` - The stack ID.
+* `deployment_names` - A set of names of the deployments in the last configuration fetched for this stack. This attribute will be empty when the resource is created and will remain empty until a configuration is fetched.
```
Comment (sebasslash) [inline website/docs/r/stack.html.markdown.md:67]: :nail_care: 
```suggestion
* `deployment_names` - The set of deployment names used in the last configuration for this stack. This attribute will be empty when the resource is created and will remain empty until a configuration is fetched.
```

**Review by sebasslash — CHANGES_REQUESTED** [review]
All in all, looks great. Really minor feedback below.

File: internal/provider/stack.go (hunk 1 above)
Comment (brandonc) [reply internal/provider/stack.go:62]: But... it's assigned on line 45. It can't be nil

**Review by sebasslash — APPROVED** [review]

**Pull request merged**